src/tiled/mainwindow.cpp

```diff
@@ -1038,6 +1041,39 @@ void MainWindow::reload()
         mDocumentManager->reloadCurrentDocument();
 }
 
+void MainWindow::executeCommand()
+{
+    Command command;
+
+    QAction *action = dynamic_cast<QAction*>(sender());
+    if (action && action->data().isValid()) {
+        //run the command passed by the action
+        command = Command::fromQVariant(action->data());
+    } else {
+        //run the default command
+        command = CommandDataModel().firstEnabledCommand();
+
+        if (!command.isEnabled) {
+            QMessageBox msgBox(window());
+            msgBox.setIcon(QMessageBox::Warning);
+            msgBox.setWindowTitle(tr("Error Executing Command"));
+            msgBox.setText(tr("You do not have any commands setup."));
+            msgBox.addButton(QMessageBox::Ok);
+            msgBox.addButton(tr("Edit commands..."), QMessageBox::ActionRole);
+            msgBox.setDefaultButton(QMessageBox::Ok);
+            msgBox.setEscapeButton(QMessageBox::Ok);
+
+            QAbstractButton *button = msgBox.buttons().last();
+            connect(button, SIGNAL(clicked()), SLOT(showCommandDialog()));
+
+            msgBox.exec();
+            return;
+        }
+    }
```

Comment:
It's not nice to duplicate all this code from the `CommandButton`. I think a `CommandManager` singleton class should be introduced, which will own a `CommandDataModel` instance and which can hold this function, to be used by the `CommandButton` and from here.